src/azfilenfs-watchdog

```diff
@@ -0,0 +1,222 @@
+            continue
+        fi
+
+        #
+        # AZ_FILES_MOUNTMAP line is of the form:
+        # account.file.preprod.core.windows.net;<stunnel_account.file.preprod.core.windows.net.conf path>;<stunnel_account.file.preprod.core.windows.net.log path>;<stunnel_account.file.preprod.core.windows.net.pid path>;<checksumHash>
+        #
+        IFS=";" read l_host l_conf l_log l_pid l_checksumhash <<< "$line"
+
+        if [ -z "$l_host" -o -z "$l_conf" -o -z "$l_pid" ]; then
+            wecho "[FATAL] Deleting invalid line in $AZ_FILES_MOUNTMAP: [$line]!"
+            l_mtime=$(ensure_az_files_mountmap_not_exist "$line")
+            [ $? -eq 0 ] && mtime_mountmap=$l_mtime
+            continue
+        fi
+
+	accept_port=$(cat $l_conf | grep accept | cut -d ':' -f 2)
+	echo "accept_port: $accept_port"
+
+        #
+        # Delete entry from AZ_FILES_MOUNTMAP if there are no mounted shares on that host.
+        # As long as we have at least one mount using the AZ_FILES_MOUNTMAP entry, we leave
+        # it around.
+        #
+        if ! echo "$findmnt" | grep "$accept_port" >/dev/null; then
+                pecho "No mounted shares for host $l_host, deleting from ${AZ_FILES_MOUNTMAP} [$line]."
+
+		# Kill stunnel process first.
+		pid=$(cat $l_pid)
+		kill -9 $pid
+		if [ $? -ne 0 ]; then
+		    eecho "[FATAL] Unable to kill stunnel process '$pid'!"
+		    exit 1
+		fi
+
+		# Cleanup stunnel files
+		rm $l_log
+		if [ $? -ne 0 ]; then
+		    eecho "[FATAL] Unable to delete stunnel log file $l_log!"
+		    exit 1
+		fi
+
+		rm $l_pid
+		if [ $? -ne 0 ]; then
+		    eecho "[FATAL] Unable to delete stunnel pid file $l_pid!"
+		    exit 1
+		fi
+
+		chattr -if $l_conf
+		rm $l_conf
+		if [ $? -ne 0 ]; then
+		    eecho "[FATAL] Unable to delete stunnel conf file $l_conf!"
+		    exit 1
+		fi
+
+                # Delete IFF mountmap is not changed since we read it above.
+                l_mtime=$(ensure_az_files_mountmap_not_exist "$line" "$mtime_mountmap")
+
+                #
+                # Update ifmatch time in case of successful updation of AZ_FILES_MOUNTMAP,
+                # so that we can distinguish between AZ_FILES_MOUNTMAP mtime changing because
+                # of our action or some mount helper changing it. In the former case
+                # it's safe to update the AZ_FILES_MOUNTMAP, so update mtime_mountmap to the
+                # mtime after this update.
+                #
+                [ $? -eq 0 ] && mtime_mountmap=$l_mtime
+                continue
+        else
+	    is_stunnel_running=$(netstat -anp | grep stunnel | grep `cat $l_pid`)
+	    echo "$is_stunnel_running"
+	    if [ -z "$is_stunnel_running" ]; then
+                vecho "stunnel is not running! Restarting the stunnel"
+
+                stunnel_status=$(stunnel $l_conf 2>&1)
+		if [ -n "$stunnel_status" ]; then
+		    eecho "[FATAL] Not able to start stunnel process for '${l_conf}'!"
+                    exit 1
```

Comment:
did you try sudo systemctl start stunnel4? instead of having seperate files and start mechanims. all the configuration in one place and just do "sudo systemctl start stunnel4"

Reply:
If the customer has stunnel.conf files in /etc/stunnel for some other purpose, then running "sudo systemctl start stunnel4" would start all stunnel processes including stunnel processes for /etc/stunnel stunnel.conf files. So I don't think running "sudo systemctl start stunnel4" is a good idea in such scenario.